subworkflows/local/map.nf

```diff
@@ -19,8 +19,26 @@ workflow MAP {
     ch_versions       = Channel.empty()
     ch_multiqc_files  = Channel.empty()
 
-    ch_input_for_mapping = reads
-                            .combine(index)
+    if ( params.mapping_tool == 'bwaaln' ) {
+        ch_index_for_mapping = index
+        ch_reads_for_mapping = reads
+
+        FASTQ_ALIGN_BWAALN ( ch_reads_for_mapping, ch_index_for_mapping )
+        ch_versions        = ch_versions.mix ( FASTQ_ALIGN_BWAALN.out.versions.first() )
+        ch_mapped_lane_bam = FASTQ_ALIGN_BWAALN.out.bam
+                                .map{
+                                    // create meta consistent with rest of workflow
+                                    meta, bam ->
+                                    new_meta = meta.clone()
+                                    new_meta.reference = meta.id_index
```

Comment:
We should not be doing clone anymore as this doesn't actually protect the metamap being modified upstream (clone doesn't actually make a true intependent copy) -  see https://www.youtube.com/watch?v=A357C-ux6Dw&t=405s&pp=ygURbmYtY29yZSBtZXRhIG1hcHM%3D for how to do this 'safely'

Reply:
Or ask me for other examples (I learnt the hard way) :)

Reply:
so this should be `new_meta = meta + [ id_index: meta.reference ]` ?

Reply:
```
new_meta = meta + [ reference: meta.id_index ]
```
I believe

Reply:
yes thanks